scope/test.h

```diff
@@ -55,6 +51,37 @@ namespace scope {
     }
   }
 
+  // eval_equal_impl for anything having std::begin and std::end overloads
+  template <typename ExceptionType, typename ExpSequenceT, typename ActSequenceT>
+  auto eval_equal_impl(ExpSequenceT&& e, ActSequenceT&& a, int, const char* const file, int line, const char* msg = "") -> decltype(std::begin(e), std::end(e), std::begin(a), std::end(a), void()) {
+    const auto abeg = std::begin(a);
+    const auto aend = std::end(a);
+    const auto ebeg = std::begin(e);
+    const auto eend = std::end(e);
+    const size_t elen = eend - ebeg;
+    const size_t alen = aend - abeg;
+
+    if (alen != elen) {
```

Comment:
I had the same thought, but I was duplicating the existing behavior. 

If the sequences have a mismatch *and* are of different lengths, and I can know only one of those, I'd prefer to know what the mismatch is---but knowing both would be even better.